### ISSUE TITLE
Allow custom implementations of InlineParsers

### ISSUE THREAD

**Issue by lalunamel** [atlassian/commonmark-java#68]
Hi there!

Continuing with the theme of [extensibility](https://github.com/atlassian/commonmark-java/issues/43#issuecomment-255873628) I'd like to be able to specify the inline parser used to parse markdown.

In my particular use case I'd like to disable a few features of markdown (e.g., inline images) and add a few of my own (e.g., `@name` will be recognized and parsed as something specific).

**Proposal:**
Add a method to `Builder`, `public Builder inlineParser(InlineParser parser)` that allows passing an implementation of `InlineParser` to `Parser`.

**e.g.,**

```
Parser.builder().inlineParser(new AtMentionParser()).build()
```

**Comment by lalunamel** [comment]
https://github.com/atlassian/commonmark-java/pull/69

**Comment by robinst** [comment]
Released in 0.9.0 with some other changes: https://github.com/atlassian/commonmark-java/releases/tag/commonmark-parent-0.9.0
